sycl/include/CL/sycl/detail/pi.h

```diff
@@ -64,6 +65,8 @@ using pi_uint64 = uint64_t;
 using pi_bool = pi_uint32;
 using pi_bitfield = pi_uint64;
 using pi_native_handle = uintptr_t;
+using pi_byte_array = std::array<std::byte, 16>;
```

Comment:
I realized that this is not a good idea to use STL in PI API. The goal was to keep it C-only such that there are no C++ compatibility issues. Let's instead have the first call to know the size, have users allocate a C-array, and have the second call to populate it.